src/dosext/mfs/lfn.c

```diff
@@ -1076,17 +1076,22 @@ static int mfs_lfn_(void)
 		if (drive < 0)
 			return drive + 2;
 
-		d_printf("LFN: %s\n", fpath);
-
 		if (_CL == 1 || _CL == 2) {
 			build_ufs_path(fpath, filename, drive);
 			if (!find_file(fpath, &st, get_redirection_root1(drive, NULL, 0), &doserrno))
 				return lfn_error(doserrno);
 			make_unmake_dos_mangled_path(filename, fpath, drive, 2 - _CL);
 		} else {
-			strupperDOS(filename);
+			char *lastslash = strrchr(filename, '\\');
+			if (lastslash) { /* Upcase but preserve the final component */
+				*lastslash = '\0';
+				strupperDOS(filename);
```

Comment:
Maybe strnupperDOS() instead?
But oh well.

Reply:
No, that was a proposal for this patch.

Reply:
Ahh, I see.